Bug fix: Avatars can't load using a HTTPS url & Give default avatar to last_post in MessageIndex.php and Subs-BoardIndex.php

HorusGoul opened the pull request.

HorusGoul committed:
Load avatar by HTTPS

Signed-off-by: Horus Lugo <horusgoul@gmail.com>

HorusGoul committed:
Load avatar by HTTPS

Signed-off by: Horus Lugo <horusgoul@gmail.com>

HorusGoul committed:
Load avatar by HTTPS

Signed-off-by: Horus Lugo <horusgoul@gmail.com>

HorusGoul committed:
Load avatar by HTTPS

Signed-off-by: Horus Lugo <horusgoul@gmail.com>

tinoest commented:
Just curious but why doesn't SMF use '//<url>' this will either be http or https depending on what your currently viewing it under. 

https://en.m.wikipedia.org/wiki/Wikipedia:Protocol-relative_URL

HorusGoul committed:
Bugfix: Load default avatar instead other user avatar.

There was a bug when you pick to not have an avatar in your profile
settings; the arrays which have a last_post propertie load an avatar
from another user, so this commit fix that.

Signed-off-by: Horus Lugo <horusgoul@gmail.com>
